noir-projects/aztec-nr/aztec/src/note/note_getter/mod.nr

```diff
@@ -103,21 +103,30 @@ pub fn get_note<Note, let N: u32, let M: u32>(
     note
 }
 
-pub fn get_notes<Note, let N: u32, let M: u32, FILTER_ARGS>(
+pub fn get_notes<Note, let N: u32, let M: u32, PREPROCESSOR_ARGS, FILTER_ARGS>(
     context: &mut PrivateContext,
     storage_slot: Field,
-    options: NoteGetterOptions<Note, N, M, FILTER_ARGS>
+    options: NoteGetterOptions<Note, N, M, PREPROCESSOR_ARGS, FILTER_ARGS>
 ) -> BoundedVec<Note, MAX_NOTE_HASH_READ_REQUESTS_PER_CALL> where Note: NoteInterface<N, M> + Eq {
     let opt_notes = get_notes_internal(storage_slot, options);
+    let preprocessed_opt_notes = apply_preprocessor(opt_notes, options.preprocessor, options.preprocessor_args);
```

Comment:
We should call the preprocessor inside `get_notes_internal`. Otherwise what we have here is a constrained function that is calling two unconstrained fns. While the fns themselves don't apply any constraints, we do need to constrain their return values. In this case, we're constraining that they do return an array of length 32 made of `Option<Note>`, which ultimately results in `blackbox::range`.

If `get_notes_internal` did the preprocessing, we'd only constrain the one value. This is not huge (~50 gates), but thought it worthwhile to point out so that we're aware of this kind of pattern.

Reply:
Interesting. Was not aware of this. Will do 🫡

Reply:
Indeed dropped it by 44 gates. Cool!